### PR TITLE
Test numa_test with thp cancels in all cases

### PR DESCRIPTION
**Pull request by Geet2**
numa_test test_thp_compare cancels even when thp is enabled on the system. Correcting the sysfs file path and approach to cancel test.
CANCEL: THP migration is not enabled on the system.

### CODE REVIEW

**Comment by Geet2** [comment]
Before :
----------
 (3/3) tests/avocado-misc-tests/memory/numa_test.py:NumaTest.test_thp_compare: STARTED
 (3/3) tests/avocado-misc-tests/memory/numa_test.py:NumaTest.test_thp_compare:  CANCEL: THP migration is not enabled on the system (0.61 s)
 
 After:
----------
 case 1 : always
 ```
# cat /sys/kernel/mm/transparent_hugepage/enabled
[always] madvise never
 (3/3) tests/avocado-misc-tests/memory/numa_test.py:NumaTest.test_thp_compare: STARTED
 (3/3) tests/avocado-misc-tests/memory/numa_test.py:NumaTest.test_thp_compare:  PASS (0.73 s)
```

 case 2 : madvise
```
# cat /sys/kernel/mm/transparent_hugepage/enabled
always [madvise] never
....
(3/3) tests/avocado-misc-tests/memory/numa_test.py:NumaTest.test_thp_compare: STARTED
(3/3) tests/avocado-misc-tests/memory/numa_test.py:NumaTest.test_thp_compare:  PASS (0.74 s)
```

 case 3 : never
```
[root@ltcrain108-lp8 avocado-fvt-wrapper]# cat /sys/kernel/mm/transparent_hugepage/enabled
always madvise [never]
 (3/3) tests/avocado-misc-tests/memory/numa_test.py:NumaTest.test_thp_compare: STARTED
 (3/3) tests/avocado-misc-tests/memory/numa_test.py:NumaTest.test_thp_compare:  CANCEL: THP migration is not enabled on the system (0.55 s)
```

File: memory/numa_test.py [hunk 1]
```diff
@@ -118,16 +119,21 @@ def test_thp_compare(self):
...
-            self.nr_pages = self.params.get(
-                'nr_pages', default=100)
-            os.chdir(self.teststmpdir)
-            self.log.info("Starting test...")
-            cmd = './bench_movepages -n %s' % self.nr_pages
-            ret = process.system(
-                cmd, shell=True, sudo=True, ignore_status=True)
-            if ret != 0:
-                self.fail('Please check the logs for failure')
+            with open(thp, 'r') as thp_file:
```
Comment (nasastry) [inline memory/numa_test.py:124]: Please use genio api here to read the file.

**Review by nasastry — CHANGES_REQUESTED** [review]

File: memory/numa_test.py (hunk 1 above)
Comment (Geet2) [reply memory/numa_test.py]: Addressed the above in latest push

**Comment by Geet2** [comment]
Test Results with above changes:
============================
Case1 :
# cat /sys/kernel/mm/transparent_hugepage/enabled
always [madvise] never

 (3/3) tests/avocado-misc-tests/memory/numa_test.py:NumaTest.test_thp_compare: STARTED
 (3/3) tests/avocado-misc-tests/memory/numa_test.py:NumaTest.test_thp_compare:  PASS (0.96 s)
 
 Case 2 :
 # cat /sys/kernel/mm/transparent_hugepage/enabled
[always] madvise never

  (3/3) tests/avocado-misc-tests/memory/numa_test.py:NumaTest.test_thp_compare: STARTED
 (3/3) tests/avocado-misc-tests/memory/numa_test.py:NumaTest.test_thp_compare:  PASS (1.92 s)
 
 Case 3:
 # cat /sys/kernel/mm/transparent_hugepage/enabled
always madvise [never]

 (3/3) tests/avocado-misc-tests/memory/numa_test.py:NumaTest.test_thp_compare: STARTED
 (3/3) tests/avocado-misc-tests/memory/numa_test.py:NumaTest.test_thp_compare:  CANCEL: THP migration is not enabled on the system (1.22 s)

**Review by PraveenPenguin — APPROVED** [review]

**Pull request merged**